OpenRA.Mods.Common/InstallUtils.cs

```diff
@@ -66,41 +90,66 @@ public static bool ExtractFromPackage(string srcPath, string package, string ann
 			Log.Write("debug", "Mounting {0}".F(package));
 			GlobalFileSystem.Mount(package, annotation);
 
-			foreach (var file in files)
+			foreach (var directory in filesByDirectory)
 			{
-				var dest = Path.Combine(destPath, file);
-				if (File.Exists(dest))
-					File.Delete(dest);
-				using (var sourceStream = GlobalFileSystem.Open(file))
-				using (var destStream = File.Create(dest))
+				var targetDir = directory.Key;
+
+				foreach (var file in directory.Value)
 				{
-					Log.Write("debug", "Extracting {0} to {1}".F(file, dest));
-					onProgress("Extracting " + file);
-					destStream.Write(sourceStream.ReadAllBytes());
+					var dest = Path.Combine(destPath, targetDir, file);
+					if (File.Exists(dest))
+					{
+						if (overwrite)
+							File.Delete(dest);
+						else
+						{
+							Log.Write("debug", "Skipping {0}".F(dest));
+							continue;
+						}
+					}
+
+					using (var sourceStream = GlobalFileSystem.Open(file))
+					using (var destStream = File.Create(dest))
+					{
+						Log.Write("debug", "Extracting {0} to {1}".F(file, dest));
+						onProgress("Extracting " + file);
+						destStream.Write(sourceStream.ReadAllBytes());
+					}
 				}
 			}
 
 			return true;
 		}
 
-		public static bool CopyFiles(string srcPath, string[] files, string destPath, Action<string> onProgress, Action<string> onError)
+		public static bool CopyFiles(string srcPath, Dictionary<string, string[]> files, string destPath,
+			bool overwrite, Action<string> onProgress, Action<string> onError)
 		{
-			foreach (var file in files)
+			foreach (var folder in files)
 			{
-				var fromPath = Path.Combine(srcPath, file);
-				if (!File.Exists(fromPath))
+				var targetDir = folder.Key;
+
+				foreach (var file in folder.Value)
 				{
-					onError("Cannot find " + file);
-					return false;
+					var fileName = file.ToLowerInvariant();
```

Comment:
You can't assume all file names are lower case. This will break on every Unix system.